lib/index.js

```diff
@@ -92,65 +104,136 @@ Rest.prototype = {
     } else if(_.isUndefined(options)) {
       options = {};
     };
-    
-    var controller = this._load(name);
+
+    var versions = options.version || options.versions || ['*'];
+    if(!Array.isArray(versions) ) { versions = [versions] };
+
+    var controller = this._load(name, versions);
 
     if(this.subfolder) {
       options.nestedPath = this.nestedPath;
     }
+    var resources = new Resources(this, name, options);
 
-    var resources = new Resources(this, name, _.merge(controller.options, options));
-  
     resources._init(controller);
 
     if('function' == typeof callback) {
       resources.parent = this;
       resources._nest(callback);
     }
-    
+
     this.addRoutes(resources.routes);
 
     return this;
   },
   mountRoutes: function(app) {
     var that = this;
-    //map the routes
     this.routes.forEach(function(route) {
       that._createRoute(route, app);
     });
-    
   },
-  _load: function(name) {
-    this._loaded = this._loaded || {};
+  _load: function(name, versions) {
+    var that = this;
+    var loaded = {};
+    var base_dir = that.settings.controllers;
+
+    _.each(versions, function(version) {
+      //Search in non searched versions plus wild card version
+      var _versions = _.clone(versions);
+      if ( _versions.indexOf("*") === -1 ) { _versions.unshift("*") }
+      _versions = _versions.splice(0, _versions.indexOf(version) + 1);
+
+      //Loop until find a folder
+      while( _versions.length > 0 ) {
```

Comment:
I could use forEach but include reverse which means twice iterations. Not a particular performance issue for this case but while and pop do the same